File: 1346_Check_If_N_And_Double_Exists.py
# we can make use of a hashmap
# lets store the double of the each array as a key inside the hashmap
# now traverse through the array and check whther there is a key which is double 
# if we found one, return true, else return false

class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        if arr[0] == 0 and arr[1] == 0:
            return True
        dictionary = {}
        for number in arr:
            dictionary[2*number] = True
        for number in arr:
            
            if number in dictionary and number != 0:
                return True
        return False
        

File: Set1_1000/1346_Check_If_N_And_Double_Exists.java
// we can make use of a hashset
// lets check the double of the element inside the set and return true, if we find one.
// else we add it into the set for later checking
// if we gets out of the loop, return false

class Solution {
    public boolean checkIfExist(int[] arr) {
        Set<Integer> set = new HashSet<>();
        for (int number: arr) {
            if (set.contains(number * 2) || (number%2 == 0 && set.contains(number / 2)))
                return true;
            set.add(number);
        }
        return false;
    }
}
